lib/services/datasets.js

```diff
@@ -0,0 +1,528 @@
+'use strict';
+
+var invariant = require('invariant'),
+  geojsonhint = require('geojsonhint/object'),
+  request = require('superagent'),
+  hat = require('hat'),
+  makeService = require('../make_service'),
+  constants = require('../constants'),
+  makeURL = require('../make_url');
+
+var Datasets = module.exports = makeService('MapboxDatasets');
+
+/**
+ * To retrieve a listing of datasets for a particular account.
+ * This request requires an access token with the datasets:read scope.
+ *
+ * @param {Function} callback called with (err, datasets)
+ * @returns {undefined} nothing, calls callback
+ * @example
+ * var mapboxClient = new MapboxClient('ACCESSTOKEN');
+ * mapboxClient.listDatasets(function(err, datasets) {
+ *   console.log(datasets);
+ *   // [
+ *   //   {
+ *   //     "owner": {account},
+ *   //     "id": {dataset id},
+ *   //     "name": {dataset name},
+ *   //     "description": {dataset description},
+ *   //     "created": {timestamp},
```

Comment:
Upstream in JSDoc, there's no support for documenting [parts of return values](http://usejsdoc.org/tags-returns.html) like you can [document parts of params](http://usejsdoc.org/tags-param.html). See previous discussion: https://github.com/documentationjs/documentation/issues/127